Source/CarthageKit/FrameworkExtensions.swift

```diff
@@ -261,6 +261,17 @@ extension NSURL {
 
 		return .Failure(.ReadFailed(self, error))
 	}
+
+	public func hasSubdirectory(possibleSubdirectory: NSURL) -> Bool {
+		guard self.scheme == possibleSubdirectory.scheme else { return false }
+
+		guard let pathComponents = self.pathComponents else { return false }
+		guard let otherPathComponents = possibleSubdirectory.pathComponents else { return false }
+
+		guard otherPathComponents.count >= pathComponents.count else { return false }
+
+		return Array(otherPathComponents[0..<pathComponents.count]) == pathComponents
```

Comment:
Can you rework this to use `if let … else …`? That should let us specify the `false` condition once instead of four times.